### PR TITLE
Hometasks1

### PR DESCRIPTION
**Pull request by pkay-14**

### CODE REVIEW

File: hometask1/hometask1-task8.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+a = [23,4,56,1,9,4,76,2]
```
Comment (MrCartoon) [inline hometask1/hometask1-task8.rb]: Change input array to ARGV(you already did it in another script).

File: hometask1/hometask1-task1.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+t = Time.now.getlocal()
```
Comment (MrCartoon) [inline hometask1/hometask1-task1.rb]: This script doesn't do what was requested in the task.

File: hometask1/hometask1-task5.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  line_num = File.foreach(file).with_index do |line, index |
+    break index + 1 if line.include? word
+  end
+  return line_num unless line_num.nil?
+end
+
+# path = File.expand_path(File.dirname(__FILE__))
+# path << filename
+
+b = get_line_number("#{__dir__}/#{filename}", line)
```
Comment (MrCartoon) [inline hometask1/hometask1-task5.rb]: Here also `__dir__` isn't required if you want to be able to specify the dynamic file's location.

File: hometask1/hometask1-task5.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+puts "line number of '#{line}' where first encountered in '#{filename}' is #{b}"
```
Comment (MrCartoon) [inline hometask1/hometask1-task5.rb]: Add another output if requested line wasn't found, cause if you run `ruby hometask1-task5.rb mytext.txt "non-exiting text"` you will get `line number of 'non-exiting text' where first encountered in 'mytext1.txt' is`.

**Comment by pkay-14** [comment]
My apologies i saw this email late. I have corrected and changed most of the request you asked. I only
have two issues. It’s task 1 and some part of task 2.

Regarding task1 i’ve made more research but i still don’t get how exactly i should write the script.

About task2, i made few changes but i’m not able to use the colour from the argv and implement it in the script. I’ve searched the internet but not finding a helpful solution yet. Can you please suggest how.

I will create another pull request to see what i’ve changed so far.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Pull request closed without merging**